### PR TITLE
Partial support for dynamic A+AAAA records on Azure 

### PR DESCRIPTION
**Pull request by viranch**
Azure DNS A/AAAA records cannot be aliased to Traffic Managers that contain other nested Traffic Managers. Due to this limitation, A/AAAA dynamic records can only do one of geo-fencing, fallback and weighted RR.

This PR implements the limited functionality, with checks in place to prevent unsupported scenarios.

### CODE REVIEW

File: octodns/provider/azuredns.py [hunk 1]
```diff
@@ -805,23 +841,58 @@ def _data_for_dynamic(self, azrecord):
...
+                geo = any(r.get('geos') for r in data['rules'])
+
+                if [rr, fallback, geo].count(True) > 1:
+                    msg = ('{} {}: A/AAAA dynamic records must use at most '
+                           'one of round-robin, fallback and geo-fencing')
+                    msg = msg.format(record.fqdn, record._type)
+                    raise AzureException(msg)
+            elif typ != 'CNAME':
+                msg = ('{}: Dynamic records in Azure must be of type '
+                       'A/AAAA/CNAME').format(record.fqdn)
```
Comment (ross) [inline octodns/provider/azuredns.py:895]: This method is getting pretty long. May be wroth breaking up some of these branches/bits into helper functions to aid readability. Doesn't have to happen right now if you don't want. It can be in a dedicated PR or something if you'd prefer (if done at all.)

**Review by ross — APPROVED** [review]
Will give this a day to hear back whether you want to break up the extra changes bit here, in another PR, or not at all.

File: octodns/provider/azuredns.py (hunk 1 above)
Comment (viranch) [reply octodns/provider/azuredns.py]: 👍 good call. updated the PR.

**Comment by ross** [comment]
You ready for a final :eye: on this or still want to tweak it more?

**Comment by viranch** [comment]
This is ready now.

File: octodns/provider/azuredns.py [hunk 2]
```diff
@@ -263,7 +269,10 @@ def _root_traffic_manager_name(record):
     # ATM names can only have letters, numbers and hyphens
     # replace dots with double hyphens to ensure unique mapping,
     # hoping that real life FQDNs won't have double hyphens
-    return record.fqdn[:-1].replace('.', '--')
+    name = record.fqdn[:-1].replace('.', '--')
+    if record._type != 'CNAME':
+        name += '-{}'.format(record._type)
```
Comment (ross) [inline octodns/provider/azuredns.py:274]: Heh. Danger of implementing one thing at a time, I guess. There are other places where I ended up having to stuff the type into the names for this same reason.

**Review by ross — APPROVED** [review]
